src/types.rs

```diff
@@ -1297,7 +1297,7 @@ fn join_bounds_inner(
     //       or the single item is of type `Trait`,
     //          and any of the internal arrays contains more than one item;
     let retry_with_force_newline = match context.config.style_edition() {
-        style_edition @ _ if style_edition <= StyleEdition::Edition2021 => {
+        style_edition if style_edition <= StyleEdition::Edition2021 => {
```

Comment:
Put `use StyleEdition::::*;` above, then 
```suggestion
         Edition2015 | Edition2018 | Edition2021 => {
```

Reply:
Done.